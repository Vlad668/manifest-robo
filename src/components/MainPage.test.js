import React from 'react';
import { Provider } from 'react-redux';
import { shallow,mount } from 'enzyme';
import MainPage from './MainPage';
import "@testing-library/jest-dom/extend-expect";
import { create } from "react-test-renderer"

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robotsReducer: [],
    searchField: '',
    isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps}/>)
 })

it('renders without crashing', () => {
  expect(create(wrapper).toJSON()).toMatchSnapshot();
});

it('fileters Robots', () => {

  const mockProps = {
    onRequestRobots: jest.fn(),
    robotsReducer: [{
          id: 1,
          name: 'Leanne Graham',
          username: 'Bret',
          email: 'Sincere@april.biz'
        }],
    searchField: 'a',
    isPending: false
  }

  const wrapper1 = shallow(<MainPage {...mockProps}/>)
  const instance = wrapper1.props()
  const i = instance.children.props.children[2].props.children[1].props.robotsPassed
  expect(i.length).toEqual(1)
  // expect(wrapper1.exists()).toEqual(1)
  
  // expect(wrapper.invoke('robotsPassed')).toEqual([]);

});

it('fileters Robots correctly', () => {
  const filteredRobots = [{
    id: 1,
    name: 'Leanne Graham',
    username: 'Bret',
    email: 'Sincere@april.biz'
  }]
  const mockProps = {
    onRequestRobots: jest.fn(),
    robotsReducer: [{
      id: 1,
      name: 'Leanne Graham',
      username: 'Bret',
      email: 'Sincere@april.biz'
    }],
    searchField: 'Leanne',
    isPending: false
  }
  const wrapper2 = shallow(<MainPage {...mockProps}/>)
  const instance = wrapper2.props()
  const i = instance.children.props.children[2].props.children[1].props.robotsPassed
  expect(i).toEqual(filteredRobots)
});

it('fileters Robots correctly 2', () => {
  const filteredRobots = [{
    id: 1,
    name: 'Leanne Graham',
    username: 'Bret',
    email: 'Sincere@april.biz'
  }]
  const mockProps = {
    onRequestRobots: jest.fn(),
    robotsReducer: [{
      id: 1,
      name: 'Leanne Graham',
      username: 'Bret',
      email: 'Sincere@april.biz'
    }],
    searchField: 'Xavier',
    isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps}/>)
  const instance = wrapper.props()
  const i = instance.children.props.children[2].props.children[1].props.robotsPassed
  expect(i).toEqual([]);
});