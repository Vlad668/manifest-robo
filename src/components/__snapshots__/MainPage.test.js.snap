// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<section
  className="tc"
>
  <h1
    className="f1"
  >
    Robo Friends
  </h1>
  <div
    className="pa2"
  >
    <input
      className="pa3 ba b--grean bg-lightest-blue"
      placeholder="Type your bot"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "border": "1px solid black",
        "height": "600px",
        "overflow": "scroll",
      }
    }
  >
    <section />
  </div>
</section>
`;
